cypress/integration/regrister.spec.js

```diff
@@ -0,0 +1,93 @@
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: No body email", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.invalidUser.noBodyemail)
+        cy.get(signUpPage.emailInput).clear().type(data.user.password)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: No @ sign email", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.invalidUser.noAtSignEmail)
+        cy.get(signUpPage.emailInput).clear().type(data.user.password)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: Invalid password", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.user.email)
+        cy.get(signUpPage.emailInput).clear().type(data.invalidUser.wrongPassword)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: Empty password", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.user.email)
+        cy.get(signUpPage.emailInput).clear().type(data.invalidUser.emptySpacePassword)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: Invalid Number of Users", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.user.email)
+        cy.get(signUpPage.emailInput).clear().type(data.user.password)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.invalidUser.invalidNumberOfUsers)
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: Checkbox not checked", () =>{
+        cy.get(signUpPage.emailInput).clear().type(data.user.email)
+        cy.get(signUpPage.emailInput).clear().type(data.user.password)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.checkbox).click()
+        cy.get(signUpPage.submitButton).click()
+    })
+
+    it("Register: Valid registration", () =>{
+        cy.get(signUpPage.emailInput).should('be.visible').clear().type(data.user.email)
+        cy.get(signUpPage.passwordInput).clear().type(data.user.password)
+        cy.get(signUpPage.numberOfUsersInput).clear().type(data.user.validNmberOfUsers)
+        cy.get(signUpPage.checkbox).click()
+        cy.get(signUpPage.submitButton).click()
+        cy.get(sidebar.myAccount.account).click({force: true})
+        cy.get(sidebar.myAccount.profile).click()
+        cy.get(navigation.logOutButton).click()
+        cy.get(loginPage.emailInput).type(data.user.email)
+        cy.get(loginPage.passwordInput).type(data.user.password)
+        cy.get(loginPage.logInButton).click()
+        cy.wait(2000)
```

Comment:
Why do we need to wait for 2 seconds? Arbitrary waits are an anti-pattern and we almost never need to use them, because you can never know how long retrieving a resource might take, it can be 3 seconds, but it can be 33 just as well. Instead, we should strive to do: 

`cy.wait('@loginData')`

and prior to that, we should intercept a login route e.g:

`cy.intercept('GET', '/login').as('loginData')`

This is, however, an advanced concept which we are yet to cover in depth.